[ENH]: Add mrdegibbs and dwibiascorrect from mrtrix3

josephmje opened the pull request:
## Summary
Addition of dwibiascorrect references  #1126

## List of changes proposed in this PR (pull-request)
- output noise map from dwidenoise
- add classes for mrdegibbs and dwibiascorrect
- remove '.nii.gz' from eddy quad folder

## Acknowledgment
I acknowledge that this contribution will be available under the Apache 2 license.

mgxd commented on nipype/interfaces/mrtrix3/preprocess.py:
```diff
@@ -75,6 +80,163 @@ class DWIDenoise(MRTrix3Base):
+    >>> import nipype.interfaces.mrtrix3 as mrt
+    >>> unring = mrt.MRDeGibbs()
+    >>> unring.inputs.in_file = 'dwi.mif'
+    >>> unring.cmdline                               # doctest: +ELLIPSIS
```
i don't think you need the `ELLIPSIS` option

mgxd commented on nipype/interfaces/mrtrix3/preprocess.py:
```diff
@@ -26,18 +26,23 @@ class DWIDenoiseInputSpec(MRTrix3BaseInputSpec):
 
 class DWIDenoiseOutputSpec(TraitedSpec):
-    out_file = File(desc="the output denoised DWI image", exists=True)
+    out_noisemap = File(desc='the output noise map', exists=True)
```
is this always output?

mgxd commented on nipype/interfaces/mrtrix3/preprocess.py:
```diff
@@ -75,6 +80,163 @@ class DWIDenoise(MRTrix3Base):
+        genfile=True)
+
+class DWIBiasCorrectOutputSpec(TraitedSpec):
+    out_bias = File(desc='the output estimated bias field')
```
is this always output?

mgxd commented on nipype/interfaces/mrtrix3/preprocess.py:
```diff
@@ -75,6 +80,163 @@ class DWIDenoise(MRTrix3Base):
+    >>> import nipype.interfaces.mrtrix3 as mrt
+    >>> bias_correct = mrt.DWIBiasCorrect()
+    >>> bias_correct.inputs.in_file = 'dwi.mif'
+    >>> bias_correct.cmdline                           # doctest: +ELLIPSIS
```
ellipsis not needed

josephmje commented:
Thanks for your feedback @mgxd! I reverted my variable name change to dwidenoise in case it affects existing code using it. Also modified the arguments to the optional output files.

mgxd commented:
tests should start passing again if you merge with master

mgxd commented on nipype/interfaces/mrtrix3/preprocess.py:
```diff
@@ -28,16 +28,17 @@ class DWIDenoiseInputSpec(MRTrix3BaseInputSpec):
         desc='set the window size of the denoising filter. (default = 5,5,5)')
     noise = File(
```
it doesn't look like this is currently being tracked as an output by the interface, but if this is set it I think `dwidenoise` would make it. You should :
1. re-add it to the OutputSpec (`noise = File(..., exists=True`)
2. add a `_list_outputs()` method that checks if `self.inputs.noise` is defined, and if so adds it to outputs.

Here's an example of (2) - though you won't have to generate a filename.

josephmje commented:
@mgxd I finished incorporating your suggestions. Let me know if there's anything else I'm missing.
